tests/test_auth.py

```diff
@@ -483,6 +483,62 @@ def test_lookup(self):
         eq_(session['user']['gender'], 'female')
 
 
+class TestRules(AuthBase):
+
+    @classmethod
+    def setUpClass(cls):
+        AuthBase.setUpClass()
+        cls.url = server.base_url + '/auth/rulesdb'
+
+    def test_default(self):
+        self.login_ok('alpha', 'alpha', check_next='/')
+        session = self.session.get(server.base_url + '/auth/session').json()
+        eq_(session['user']['gender'], 'male')
+
+        self.login_ok('beta', 'beta', check_next='/')
+        session = self.session.get(server.base_url + '/auth/session').json()
+        eq_(session['user']['gender'], 'female')
+        eq_(session['user']['team'], 'Gramex')
+
+        # When selector is a nonexistent attribute, nothing changes
+        self.login_ok('gamma', 'gamma', check_next='/')
+        session = self.session.get(server.base_url + '/auth/session').json()
+        eq_(session['user']['gender'], 'female')
+
+        # Check if the empty string match for Gamma works
+        eq_(session['user']['email'], 'gamma@null.com')
```

Comment:
Add test case where we can add field="expiry" with a value= of date type. Also a field="cookie_expires" with value= of float type

Reply:
Please also test that no user has an email none@none.com nor gamma@null.com.

Also test that team=Gramex for all